libgnucash/engine/gnc-budget.cpp

```diff
@@ -781,8 +781,7 @@ get_perioddata (const GncBudget *budget, const Account *account, guint period_nu
                 data.value = get_account_period_value (budget, account, i);
             vec.emplace_back (data);
         }
-        map->insert_or_assign (account, vec);
-        return vec[period_num];
+        map_iter = map->emplace(account, vec).first;
```

Comment:
Nothing weird here, clang just does a better job of recognizing that you're trying to return a reference to something that won't be valid once the function exits. It's always nicer to get a compile error than a crash!

`insert` instead of `emplace`: The latter is for constructing a new element from the element type's constructor arguments, something you can't do in this case. Same for creating the vector element: If you construct the vector yourself use `push_back`; in this case you *could* do
```
   auto value_set = s_account_period_value_set(budget, account, i)
   vec.emplace_back(get_account_period_note(budget, account, i),
                                    is_set, is_set ? get_account_period_value(budget, account, i) : gnc_numeric_zero());
```
When you construct the value separately you should pass it wrapped with `std::move`, e.g. `map_iter = map->insert(account, std::move(vec)).first;` to transfer `vec`'s contents instead of copying them.